pvlib/iotools/tmy.py

```diff
@@ -181,8 +176,9 @@ def read_tmy3(filename, coerce_year=None, recolumn=True):
     data_ymd = pd.to_datetime(data['Date (MM/DD/YYYY)'], format='%m/%d/%Y')
     # shift the time column so that midnite is 00:00 instead of 24:00
     shifted_hour = data['Time (HH:MM)'].str[:2].astype(int) % 24
-    # shift the dates at midnite so they correspond to the next day
-    data_ymd[shifted_hour == 0] += datetime.timedelta(days=1)
+    # shift the dates at midnight (24:00) so they correspond to the next day.
+    # If midnight is specified as 00:00 do not shift date.
+    data_ymd[data['Time (HH:MM)'].str[:2] == '24'] += datetime.timedelta(days=1)  # noqa: E501
```

Comment:
These lines contain the changes to the function itself. The remaining changes are only related to the formatting of the documentation.